### PR TITLE
Update calculations and tests

### PR DESCRIPTION
**Pull request by sydneyshile**
Using table-driven tests to avoid duplicate code.

### CODE REVIEW

File: difference.go [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+If result is positive, it means there is an increase.
+If result is negative, it means there is a decrease.
+Initial value cannot be zero.
 */
-func PercentChange(initial float64, final float64) float64 {
-	diff := math.Abs(final - initial)
-
-	return diff / initial * 100
+func PercentChange(initial float64, final float64) (float64, error) {
+	if initial == 0 {
```
Comment (automoto) [inline difference.go:16]: I think we can check for above zero here instead like `>0`, please update this

Comment (automoto) [reply difference.go:16]: eventually we will want to be able to handle zero here but we can add it later

Comment (sydneyshile) [reply difference.go:16]: PercentChange should be able to handle all values except zero value. There is no percent change for zero value. 
If zero is denominator, the answer is infinity. It does not make sense mathematically.

Comment (automoto) [reply difference.go:16]: I see, so negative numbers are ok?

**Review by automoto — APPROVED** [review]

File: difference.go (hunk 1 above)
Comment (sydneyshile) [reply difference.go:16]: I think negative numbers should be fine. We can calcualate percent changes with negative value as we take absolute of initial value as denominator.  I use this link https://www.calculatorsoup.com/calculators/algebra/percent-change-calculator.php to check calcualation.

**Pull request merged**